hpx/runtime/actions/basic_action.hpp

```diff
@@ -897,6 +970,28 @@ namespace hpx { namespace serialization
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////
+#if defined(HPX_HAVE_ITTNOTIFY) && !defined(HPX_HAVE_APEX)
+#define HPX_REGISTER_ACTION_DECLARATION_NO_DEFAULT_GUID(action)               \
+    namespace hpx { namespace actions { namespace detail {                    \
+        template <> HPX_ALWAYS_EXPORT                                         \
+        char const* get_action_name<action>();                                \
+        template <> HPX_ALWAYS_EXPORT                                         \
+        util::itt::string_handle const& get_action_name_itt<action>();        \
```

Comment:
Can we just factor the `get_action_name_itt` declaration out? this would avoid the duplication of the code in those macros.